src/coreclr/vm/object.h

```diff
@@ -638,8 +638,20 @@ class ArrayBase : public Object
 
     inline static unsigned GetBoundsOffset(MethodTable* pMT);
     inline static unsigned GetLowerBoundsOffset(MethodTable* pMT);
+
+    template<typename T> friend struct ::cdac_offsets;
 };
 
+#ifndef DACCESS_COMPILE
+template<>
+struct cdac_offsets<ArrayBase>
+{
+    static constexpr size_t m_NumComponents = offsetof(ArrayBase, m_NumComponents);
+
+    static constexpr INT32* s_arrayBoundsZero = &ArrayBase::s_arrayBoundsZero;
```

Comment:
I think we should rename it.  in https://github.com/dotnet/runtime/pull/104999 Jeremy is adding some `cdac_offsets<Foo>::NestedStructSize` constants because `Foo` has a private `NestedStruct`. 

We could just name it `cdac<T>`  or `data_contract<T>`